app.rb

```diff
@@ -0,0 +1,112 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'books'
+require_relative 'rental'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'List of all books:'
+    @books.each { |book| puts "#{book.title} written by: #{book.author}" }
+  end
+
+  def list_all_people
+    puts 'List of people:'
+    @people.each do |person|
+      if person.is_a?(Student)
+        puts "[Student] ID: #{person.id}, Name: #{person.name}, Age: #{person.age}, Classroom: #{person.classroom}"
+      elsif person.is_a?(Teacher)
+        puts "[Teacher] ID: #{person.id}, Name: #{person.name}, Age: #{person.age},
+        Specialization: #{person.specialization}"
+      else
+        puts 'Invalid either student or teacher please'
+      end
+    end
+  end
+
+  def create_a_person # rubocop:todo Metrics/MethodLength
+    print 'Enter your name: '
+    name = gets.chomp
+    print 'Enter your Age: '
+    age = gets.chomp.to_i
+
+    print "Are you a student or a teacher? (Enter 'student' or 'teacher'): "
```

Comment:
- [ ] Hey there! 🌟 Please focus on the provided console. When initiating the person creation process, the user should be prompted to choose between creating a student or a teacher. The input for this selection should be done using numbers, not strings. 🤖 Kindly modify your "create person" function to align with this console requirement. Remember, the app should mirror the behavior outlined in the project requirements [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) 🚀

```sh
 # *Create a person (teacher or student, not a plain `Person`)
  def create_person
    print "Do you want to create a student (1) or a teacher (2)? [Input the number]: "
    # *If needed, ask for parameters for the option.
    person_type = gets.chomp
    if person_type != "1" && person_type != "2"
      puts "Invalid option"
      return
    end

    print "Age: "
    # *If needed, ask for parameters for the option.
    age = gets.chomp

    print "Name: "
    # *If needed, ask for parameters for the option.
    name = gets.chomp

    person =
      if person_type == "1"
        print "Has parent permission? [Y/N]: "
        # *If needed, ask for parameters for the option.
        parent_permission = gets.chomp
        parent_permission = parent_permission.downcase == "y"

        Student.new(age, name, parent_permission)
      elsif person_type == "2"
        print "Specialization: "
        # *If needed, ask for parameters for the option.
        specialization = gets.chomp

        Teacher.new(age, specialization, name)
      end

    @people << person
    puts "Person created successfully"
  end
```